backend/app/recommender.py

```diff
@@ -41,17 +41,48 @@ def _apply_hard_filters(
     """Applies hard filters to the list of stalls."""
     filtered_stalls = stalls
 
-    if preferences.get("is_ada"):
+    if preferences.get("ada"):
         filtered_stalls = [s for s in filtered_stalls if s.features.is_ada]
 
-    if preferences.get("is_ev"):
+    if "size" in preferences:
+        size_prefs = preferences["size"]
+        if not isinstance(size_prefs, list):
+            size_prefs = [size_prefs]
+
+        for size_pref in size_prefs:
+            if size_pref.startswith("not_"):
+                size_to_exclude = size_pref.split("not_")[1]
+                if size_to_exclude in size_map:
+                    size_class_to_exclude = size_map[size_to_exclude]
+                    filtered_stalls = [
+                        s for s in filtered_stalls
+                        if s.features.width_class != size_class_to_exclude
+                    ]
+
+
+    if preferences.get("ev"):
         filtered_stalls = [s for s in filtered_stalls if s.features.is_ev]
+    elif preferences.get("ev") is False:
+        filtered_stalls = [s for s in filtered_stalls if not s.features.is_ev]
+
+    if "connector" in preferences:
+        connector_prefs = preferences["connector"]
+        if not isinstance(connector_prefs, list):
+            connector_prefs = [connector_prefs]
+
+        for conn_pref in connector_prefs:
+            if conn_pref.startswith("no_"):
+                conn_to_exclude = conn_pref.split("no_")[1]
+                filtered_stalls = [
+                    s for s in filtered_stalls
+                    if not s.features.connectors or conn_to_exclude not in s.features.connectors
```

Comment:
I have a suggestion that might help this works smarter so you can try:
```suggestion
                    if s.features.connectors and conn_to_exclude not in s.features.connectors.lower()
```